### PR TITLE
Add up button at the end of events on front page

### PR DESCRIPTION
**Pull request by magnusbrecke**
# Description

Added a up button to the bottom of the shown events on front page. When clicked the button takes user to top of front-page and sets the shown events and articles to standard. 

# Result
Before:

https://github.com/webkom/lego-webapp/assets/145492323/4d4851c8-339c-4a22-ba1e-efb7b30b5e25

After:

https://github.com/webkom/lego-webapp/assets/145492323/490a48b5-bf6a-4507-be64-0e56da8f7a86

If you've made visual changes, please check the boxes below and include images showing the changes. Descriptions are appreciated.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

- [x] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

### CODE REVIEW

File: app/routes/overview/components/Overview.tsx [hunk 1]
```diff
@@ -96,6 +96,18 @@ const Overview = () => {
...
+  const scrollToTop = () => {
+    window.scrollTo({ top: 0, behavior: 'smooth' });
+
+    setTimeout(function () {
+      setArticlesToShow(ARTICLES_TO_SHOW);
+    }, 200);
+
+    setTimeout(function () {
+      setEventsToShow(EVENTS_TO_SHOW);
+    }, 200);
```
Comment (ivarnakken) [inline app/routes/overview/components/Overview.tsx:108]: Why are these here? Not sure how they are related

Comment (Bestem0r) [inline app/routes/overview/components/Overview.tsx:108]: You can move these two into a single one

File: app/routes/overview/components/Overview.tsx [hunk 2]
```diff
@@ -131,6 +143,12 @@ const Overview = () => {
           <Icon onClick={showMore} name="chevron-down-outline" size={30} />
         </div>
       )}
+
+      {frontpage.length > 8 && events.length < eventsToShow && (
+        <div className={styles.showMore}>
+          <Icon onClick={scrollToTop} name="chevron-up-outline" size={30} />
+        </div>
+      )}
```
Comment (Bestem0r) [inline app/routes/overview/components/Overview.tsx:151]: You can merge these two checks into a single one using the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator).

**Review by Bestem0r — APPROVED** [review]
Really cool feature!

File: app/routes/overview/components/Overview.tsx [hunk 3]
```diff
@@ -131,6 +143,12 @@ const Overview = () => {
...
     </Container>
```
Comment (ivarnakken) [inline app/routes/overview/components/Overview.tsx:152]: This is a good example of when to use a [turnary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator), since the "condition" here is the opposite of the one above.

```ts
      {frontpage.length > 8 && events.length > eventsToShow ? (
        <div className={styles.showMore}>
          <Icon onClick={showMore} name="chevron-down-outline" size={30} />
        </div>
      ) : (
        <div className={styles.showMore}>
          <Icon onClick={scrollToTop} name="chevron-up-outline" size={30} />
        </div>
      )}
```

You can even take it a step further and say reuse the wrapper if you want

```ts
        <div className={styles.showMore}>
          {frontpage.length > 8 && events.length > eventsToShow ? (
            <Icon onClick={showMore} name="chevron-down-outline" size={30} />
           ) : (
            <Icon onClick={scrollToTop} name="chevron-up-outline" size={30} />
          )}
        </div>
```

**Review by ivarnakken — CHANGES_REQUESTED** [review]
Very cool and useful feature! 😄

File: app/routes/overview/components/Overview.tsx (hunk 1 above)
Comment (magnusbrecke) [reply app/routes/overview/components/Overview.tsx:108]: I think they kicked in somehow before the scroll function which led to not reaching the top when applying the scroll function but the bottom of the reduced event list. Adding the timeout fixed the problem

[the remark this answers is not in the document]
Comment (Arashfa0301) [reply app/routes/overview/components/Overview.tsx:108]: Agreed. Perhaps avoid using outdated features.

Comment (norbye) [reply app/routes/overview/components/Overview.tsx:108]: Could you try without these? When I'm running it locally it worked just fine without

Comment (norbye) [reply app/routes/overview/components/Overview.tsx:108]: Or what environment did you use where it didn't work? if they are indeed necessary it'd be nice with @Bestem0r 's suggestion of adding them to the same setTimeout function, and maybe adding a comment as to why they are required as it doesn't feel super intuitive
```suggestion
    setArticlesToShow(ARTICLES_TO_SHOW);
    setEventsToShow(EVENTS_TO_SHOW);
```
or
```suggestion
    // This timeout is needed because of bla bla bla
    setTimeout(() => {
      setArticlesToShow(ARTICLES_TO_SHOW);
      setEventsToShow(EVENTS_TO_SHOW);
    }, 200);
```

[the remark this answers is not in the document]
Comment (haukkagu) [reply app/routes/overview/components/Overview.tsx:108]: Ordinary functions aren't "outdated". In this case, it doesn't really matter if you use ordinary or arrow functions. However, some argue that arrow functions look nicer.

**Review by jonasdeluna — APPROVED** [review]
Great work!

File: app/routes/overview/components/Overview.tsx (hunk 1 above)
Comment (magnusbrecke) [reply app/routes/overview/components/Overview.tsx]: Yeah I tried both in chrome and safari and for me it still doesn't work without the timeout, however I have now put them into one single timeout, and I added a comment to explain why they are needed

**Review by norbye — APPROVED** [review]

**Review by haukkagu — APPROVED** [review]
Awesome :star_struck:

**Comment by ivarnakken** [comment]
Are we still sure the timeouts are needed? It works fine without on my computer 🤔 Tested on both chrome and safari

**Comment by magnusbrecke** [comment]
> Are we still sure the timeouts are needed? It works fine without on my computer 🤔 Tested on both chrome and safari


https://github.com/webkom/lego-webapp/assets/145492323/cfb93c28-4521-41f7-9c4b-056aff699ec4

@ivarnakken this is what happens on my computer when they are not in timeout function, however tested in chrome again as well and it worked fine this time around.

**Comment by ivarnakken** [comment]
> @ivarnakken this is what happens on my computer when they are not in timeout function, however tested in chrome again as well and it worked fine this time around.

Are you able to reproduce it or does it just happen "randomly"?

**Comment by magnusbrecke** [comment]
> > @ivarnakken this is what happens on my computer when they are not in timeout function, however tested in chrome again as well and it worked fine this time around.
> 
> Are you able to reproduce it or does it just happen "randomly"?

This happens for me every single time in safari

**Comment by ivarnakken** [comment]
> This happens for me every single time in safari

Okay - very strange - but I can live it it 😄

**Review by ivarnakken — APPROVED** [review]

**Review by eikhr — APPROVED** [review]
Cool!
You will have to rebase this because I renamed the `Overview`-component, but I think all your code will still work without changes:)

**Review by falbru — APPROVED** [review]
Very good job! You're one handsome man!!!
